File: DSA/DSA py and go/stack and queue/queue.py
class Queuesarray:

    def __init__(self):
        self.data = []
    
    def __len__(self):
        return len(self.data)
    
    def isempty(self):
        return len(self.data) == 0
    
    def enqueue(self,e):
        return self.data.append(e)

    def dequeue(self):
        if self.isempty():
            print("Queue is empty")
            return
        return self.data.pop(0)
    
    def first(self):
        if self.isempty():
            print("Queue is empty")
            return
        return self.data[0]

Q = Queuesarray()
Q.enqueue(10)
Q.enqueue(20)
Q.enqueue(30)
Q.enqueue(40)
Q.enqueue(50)
print(Q.dequeue())
print(Q.data)
print(Q.first())

File: DSA/DSA py and go/stack and queue/queue.go
package main

import "fmt"

type Queuearray struct {
	data []int
	size int
}

func (QA *Queuearray) Len() int {
	return len(QA.data)
}

func (QA *Queuearray) enqueue(e int) {
	QA.data = append(QA.data, e)
}

func (QA *Queuearray) dequeue() (b int) {
	if QA.Len() == 0 {
		fmt.Println("Queue is empty")
		return
	} else {
		b := QA.data[0]
		QA.data = QA.data[1:QA.Len()]
		return b
	}
}

func (QA *Queuearray) first() (b int) {
	if QA.Len() == 0 {
		fmt.Println("Queue is empty")
		return
	} else {
		b := QA.data[0]
		return b
	}
}

func main() {
	Q := Queuearray{}
	Q.enqueue(10)
	Q.enqueue(20)
	Q.enqueue(30)
	Q.enqueue(40)
	Q.enqueue(50)
	fmt.Println(Q.data)
	fmt.Println(Q.dequeue())
	fmt.Println(Q.data)
	fmt.Println(Q.first())
}
